src/main/scala/firrtl/stage/FirrtlCompilerModes.scala

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package firrtl.stage
+
+import firrtl.transforms._
+import firrtl.passes.memlib._
+import firrtl.options.{ShellOption, HasShellOptions}
+
+object OptimizeForFPGA extends HasShellOptions {
+  private val fpgaAnnos = Seq(
+    InferReadWriteAnnotation,
+    RunFirrtlTransformAnnotation(new InferReadWrite),
+    DefaultReadFirstAnnotation,
+    RunFirrtlTransformAnnotation(new SetDefaultReadUnderWrite),
+    RunFirrtlTransformAnnotation(new SimplifyMems),
+    PassthroughSimpleSyncReadMemsAnnotation
+  )
+   val options = Seq(
+     new ShellOption[Unit](
+       longOption = "optimize-for-fpga",
+       toAnnotationSeq = a => fpgaAnnos,
+       helpText = "Choose compilation strategies that generally favor FPGA targets",
+       shortOption = Some("fpga"),
+     )
```

Comment:
This is a bit bikesheddy, but I proposed a CLI scoping scheme in https://github.com/chipsalliance/firrtl/pull/1869 that we could maybe follow here. Maybe the long option should be something like `--backend:fpga`?

Reply:
I'll bikeshed this... Maybe `--target:<asic,fpga>` or `--circuit:<asic,fpga>` (since ASIC's and FPGAs are both integrated circuits). `--backend` makes me think of "compiler back end" which makes me think "VHDL" or "Verilog". 🤷‍♀️

Reply:
I like `--target:` since it would also work for `simulation`.
So: `--target:<asic,fpga,sim>`

Reply:
I made it `--target:fpga`. Note that this is the name of the flag, not a value. This follows the `--pretty:` flag @jackkoenig added earlier, but it might make less sense in a mutually-exclusive context (like target) than in a setting where the flags within the "scope" are orthogonal (like `--pretty:`). Maybe this is an opportunity for a later upstream improvement.